tk_toolchain/cmd_line_tools/tk_config_update/__init__.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Toolkit Configuration Update
+
+Update the version of a bundle in a config to the specified version and pushes
+it back to the source repository.
+
+Usage:
+    tk-config-update <config> <bundle> <version> [--push-changes]
+
+Options:
+    --push-changes  Pushes the changes to the repository. If not specified,
+                    the remote repository is not updated.
+
+Example:
+    tk-config-update git@github.com:shotgunsoftware/tk-config-default2.git tk-core v0.19.0
+"""
+
+import subprocess
+import sys
+import os
+import shutil
+import atexit
+import tempfile
+import docopt
+
+try:
+    from ruamel import yaml
+except ImportError:
+    sys.exit("This tool can only be run with Python 2.7 or greater.")
+
+
+# FIXME: Maybe we should rename the other repository class (tk_toolchain.repo.Repository)
```

Comment:
I considered it, but it would require quite a refactor and I want to keep this branch simple.